### PR TITLE
[CPCN-27] ui: Support hook for modifying versions label text

### PR DESCRIPTION
**Pull request by MarkLark86**
## Checklist

- [x] This pull request is not adding new forms that use redux
- [x] This pull request is adding missing TypeScript types to modified code segments where it's easy to do so with confidence
- [x] This pull request is replacing `lodash.get` with optional chaining for modified code segments

### CODE REVIEW

File: assets/index.ts [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+export function startApp(): Promise<INewshubApi> {
+    if (started === true) {
+        return Promise.resolve(newshubApi);
+    }
+
+    started = true;
+    return Promise.resolve(newshubApi);
+}
+
+setTimeout(() => {
```
Comment (petrjasek) [inline assets/index.ts:19]: I'm wondering what's the usecase for this startApp to be called? just to set the `started` flag?

File: assets/api.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {IArticle} from './interfaces';
+import {gettext} from './utils';
+
+export interface INewshubApi {
```
Comment (petrjasek) [inline assets/api.ts:4]: maybe we should call it like `INewshubExtensionApi` if we expect it to be overriden, I would expect `INewshubApi` something I can use but not change

File: assets/api.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        wire: {
+            getVersionsLabelText(item: IArticle, plural?: boolean): string;
+        };
+    };
+    localization: {
+        gettext(message: string, params?: {[key: string]: string | number}): string;
+    };
+}
+
+function getVersionsLabelText(_item: IArticle, plural?: boolean): string {
```
Comment (petrjasek) [inline assets/api.ts:15]: I'm wondering if we should have rather the function return some predefined options like:
```ts
getAncestorType(_item: IArticle): 'versions' | 'sequence';
```
and handle those options in core, that would make translations etc simpler

Comment (MarkLark86) [reply assets/api.ts:15]: @petr to keep it simple for now, I'll move this code into core. We can move it out later if desired.

**Pull request closed without merging**